mscore/instrwidget.cpp

```diff
@@ -478,6 +480,139 @@ void InstrumentsWidget::genPartList(Score* cs)
+                  _partiturSelectedItems.append(item);
+            }
+      else {
+            for (QTreeWidgetItem* parent : selectedParents) {
+                  _partiturSelectedItems.append(parent);
+                  parent->setSelected(true);
+                  for (int childId = 0; childId < parent->childCount(); childId++)
+                        parent->child(childId)->setSelected(true);
+                  
+                  }
+            }
+      }
+
+//---------------------------------------------------------
+//   movePartiturListItems
+//---------------------------------------------------------
+
+void InstrumentsWidget::movePartiturListItems(bool moveUp)
+      {
+            if (_partiturSelectionType == NONE_SELECTED) return;
+            
+            int delta = moveUp ? -1 : 1;
+            QList<QTreeWidgetItem*> originalSelection = partiturList->selectedItems();
+            QList<QTreeWidgetItem*> itemsToMove = QList<QTreeWidgetItem*>(_partiturSelectedItems);
+            PartiturSelectionType selectionType = _partiturSelectionType;
+            partiturList->selectionModel()->clear();
+            
+            QTreeWidgetItem* item;
+            if (selectionType == PARENTS_SELECTED) {
+                  int idx = delta > 0 ? partiturList->topLevelItemCount() - 1 : 0;
+                  for ( ; (item = partiturList->topLevelItem(idx)); idx += -delta) {
+                        if (!itemsToMove.contains(item))
+                              continue;
+                        
+                              bool isExpanded = partiturList->isItemExpanded(item);
+                              int idx = partiturList->indexOfTopLevelItem(item);
+                              int n = partiturList->topLevelItemCount();
+                              // if part not last, move one slot down
+                              if ((idx + delta < n) && (idx + delta >= 0)) {
+                                    QTreeWidgetItem* item1 = partiturList->takeTopLevelItem(idx);
+                                    // Qt looses the QComboBox set into StaffListItem's when they are re-inserted into the tree:
+                                    // get the currently selected staff type of each combo and re-insert
+                                    int numOfStaffListItems = item1->childCount();
+#if (!defined (_MSCVER) && !defined (_MSC_VER))
+                                    int staffIdx[numOfStaffListItems];
+#else
+                                    // MSVC does not support VLA. Replace with std::vector. If profiling determines that the
+                                    //    heap allocation is slow, an optimization might be used.
+                                    std::vector<int> staffIdx(numOfStaffListItems);
+#endif
+                                    int itemIdx;
+                                    for (itemIdx=0; itemIdx < numOfStaffListItems; ++itemIdx)
```

Comment:
Space around operator as per coding rules: `itemIdx = 0`

Reply:
I have imported the project in QT Creator to resolve spacing and indentation issues. A lot of issues were already there. I missed a few in my own addition as XCode refuses to indent if it does not expect an indentation. Working on a few of the other comments now.